packages/typespec-ts/test/modularUnit/static/urlTemplate.spec.ts

```diff
@@ -0,0 +1,474 @@
+        assert("{&keys}", "&keys=semi,%3B,dot,.,comma,%2C");
+      });
+
+      it("variable list explode", () => {
+        assert("{list*}", "red,green,blue");
+        assert("{+list*}", "red,green,blue");
+        assert("{#list*}", "#red,green,blue");
+        assert("{/list*}", "/red/green/blue");
+        assert("{;list*}", ";list=red;list=green;list=blue");
+        assert("{.list*}", ".red.green.blue");
+        assert("{?list*}", "?list=red&list=green&list=blue");
+        assert("{&list*}", "&list=red&list=green&list=blue");
+      });
+
+      it("variable associative array explode", () => {
+        assert("{+keys*}", "semi=;,dot=.,comma=,");
+        assert("{#keys*}", "#semi=;,dot=.,comma=,");
+        assert("{/keys*}", "/semi=%3B/dot=./comma=%2C");
+        assert("{;keys*}", ";semi=%3B;dot=.;comma=%2C");
+        assert("{?keys*}", "?semi=%3B&dot=.&comma=%2C");
+        assert("{&keys*}", "&semi=%3B&dot=.&comma=%2C");
+      });
+
+      it("encodes associative arrays correctly", () => {
+        assert("{chars*}", "%C3%BC=%C3%BC");
+      });
+    });
+  });
+
+  describe("apiVersion", () => {
+    it("should expand apiVersion", () => {
+      const assert = createAssertion({
+        "api-version": "2023-05-01.17.0",
+        timeOut: undefined
+      });
+      assert(
+        "/pools{?api-version,timeOut}",
+        "/pools?api-version=2023-05-01.17.0"
+      );
+    });
+  });
+
+  describe("allowReserved option", () => {
+    describe("normal path", () => {
+      it("should not encode reserved characters if enable allowReserved", () => {
+        const assert = createAssertion(
+          {
+            path: "/foo/bar",
+            query: "bar,baz"
+          },
+          { allowReserved: true }
+        );
+        assert("{path}/here{?query}", "/foo/bar/here?query=bar,baz");
+      });
+
+      it("should encode reserved characters if disable allowReserved", () => {
+        const assert = createAssertion(
+          {
+            path: "/foo/bar",
+            query: "bar,baz"
+          },
+          { allowReserved: false }
+        );
+        assert("{path}/here{?query}", "%2Ffoo%2Fbar/here?query=bar%2Cbaz");
+      });
+    });
+    describe("reserved path", () => {
+      it("should not encode reserved characters if enable allowReserved", () => {
+        const assert = createAssertion(
+          {
+            path: "/foo/bar",
+            query: "bar,baz"
+          },
+          { allowReserved: true }
+        );
+        assert("{+path}/here{?query}", "/foo/bar/here?query=bar,baz");
+      });
+
+      it("should encode reserved characters if disable allowReserved", () => {
+        const assert = createAssertion(
+          {
+            path: "/foo/bar",
+            query: "bar,baz"
+          },
+          { allowReserved: false }
+        );
+        assert("{+path}/here{?query}", "%2Ffoo%2Fbar/here?query=bar%2Cbaz");
+      });
+    });
+  });
+
+  describe("path parameters", () => {
```

Comment:
is there any query parameters and header parameters uts?

Reply:
we have rich cadl-ranch cases for path and query and header is not supported yet.